tmt/steps/__init__.py

```diff
@@ -672,7 +711,36 @@ def _patch_raw_datum(raw_datum: _RawStepData, incoming_raw_datum: _RawStepData)
                         if raw_datum['name'] != needle:
                             continue
 
-                        _patch_raw_datum(raw_datum, incoming_raw_datum)
+                        _patch_raw_datum(raw_datum, incoming_raw_datum, invocation)
+
+                        break
+
+                    else:
+                        raise GeneralError(
+                            f"Cannot update phase '{needle}', no such name was found.")
+
+                else:
+                    debug('  needle-less update (postponed)')
+
+                    postponed_invocations.append(invocation)
+
+            elif invocation.options.get('update_missing'):
```

Comment:
So this `elif` branch is identical with the previous one except for the debug mesage and `missing_only` parameter. Would it make sense to join them into a single one to prevent the copy & paste here?

Reply:
Good point, although I'd leave it for a future patch. I plan to add `--remove`, which will have its own code doing magic, and I expect this to reveal more places where things could be unified and simplified.

Reply:
Ack.